src/hermes/commands/deposit/invenio_rdm.py

```diff
@@ -0,0 +1,717 @@
+        # this can be removed as it defaults to today.
+        "publication_date": date.today().isoformat(),
+        "title": metadata["name"],
+        "creators": creators,
+        # TODO: Use a real description here. Possible sources could be
+        # `tool.poetry.description` from pyproject.toml or `abstract` from
+        # CITATION.cff. This should then be stored in codemeta description field.
+        "description": metadata["name"],
+        "access_right": access_right,
+        "license": license,
+        "embargo_date": embargo_date,
+        "access_conditions": access_conditions,
+        # TODO: If a publisher already has assigned a DOI to the files we want to
+        # upload, it should be used here. In this case, Invenio will not give us a new
+        # one. Set "prereserve_doi" accordingly.
+        "doi": None,
+        # This prereserves a DOI that can then be added to the files before publishing
+        # them.
+        # TODO: Use the DOI we get back from this.
+        "prereserve_doi": True,
+        # TODO: A good source for this could be `tool.poetry.keywords` in pyproject.toml.
+        "keywords": None,
+        "notes": None,
+        "related_identifiers": None,
+        # TODO: Use `contributors`. In the case of the hermes workflow itself, the
+        # contributors are currently all in `creators` already. So for now, we set this
+        # to `None`. Change this when relationship between authors and contributors can
+        # be specified in the processing step.
+        "contributors": None,
+        "references": None,
+        "communities": communities,
+        "grants": None,
+        "subjects": None,
+        "version": metadata.get('version'),
+    }.items() if v is not None}
+
+    return deposition_metadata
+
+
+def _get_license_identifier(ctx: CodeMetaContext, license_api_url: str):
+    """Get Invenio license representation from CodeMeta.
+
+    The license to use is extracted from the ``license`` field in the
+    :class:`CodeMetaContext` and converted into an appropiate license identifier to be
+    passed to an Invenio instance.
+
+    A license according to CodeMeta may be a URL (text) or a CreativeWork. This function
+    only handles URLs. If a ``license`` field is present in the CodeMeta and it is not
+    of type :class:`str`, a :class:`RuntimeError` is raised.
+
+    Invenio instances take a license string which refers to a license identifier.
+    Typically, Invenio instances offer licenses from https://opendefinition.org and
+    https://spdx.org. However, it is possible to mint PIDs for custom licenses.
+
+    An API endpoint (usually ``/api/vocabularies/licenses``) can be used to check whether a given
+    license is supported by the Invenio instance. This function tries to retrieve the
+    license by the identifier at the end of the license URL path. If this identifier
+    does not exist on the Invenio instance, a :class:`RuntimeError` is raised. If no
```

Comment:
This needs updating. Now, other things are checked before giving up.

Reply:
Yes, unfortunate they use the different URLs. However, it's at least documented... even though I could not find the source 😅